[Bug]: CI is broken on main — ruff format and test_refs failures

## Describe the Bug

CI on `main` has been failing for the last 8 runs (since 2026-04-17). Three unrelated issues:

1. Ruff format fails on two browser tool files
2. `tools/tests/test_refs.py` has two failing tests due to a regression in the browser refs system
3. `core/tests/test_model_catalog.py` has a failing assertion against the OpenRouter frontier model list

## To Reproduce

1. Lint: `uv run --project core ruff format --check core/ tools/` fails on:
   - `tools/src/gcu/browser/tools/inspection.py`
   - `tools/src/gcu/browser/tools/interactions.py`

2. Tools tests: `uv run pytest tools/tests/test_refs.py -v` fails on:
   - `test_skips_structural_roles`
   - `test_named_content_roles_get_refs`

3. Framework tests: `uv run pytest core/tests/test_model_catalog.py -v` fails on:
   - `test_openrouter_catalog_tracks_current_frontier_set`

## Expected Behavior

CI on `main` should be green.

## Environment

- CI: GitHub Actions (ubuntu-latest)
- Python: 3.11
- ruff: 0.15.0

## Logs

```
Lint:
Would reformat: tools/src/gcu/browser/tools/inspection.py
Would reformat: tools/src/gcu/browser/tools/interactions.py

Tools:
FAILED tests/test_refs.py::TestAnnotateSnapshot::test_skips_structural_roles
  AssertionError: assert 'navigation' not in {'button', 'heading', 'link', 'navigation', 'textbox'}
FAILED tests/test_refs.py::TestAnnotateSnapshot::test_named_content_roles_get_refs
  AssertionError: assert 'img' in {'button', 'heading', 'link', 'navigation', 'textbox'}

Framework:
FAILED tests/test_model_catalog.py::test_openrouter_catalog_tracks_current_frontier_set
  AssertionError: assert ['openai/gpt-....6-plus', ...] == ['openai/gpt-...eepseek-v3.2']
```

## Additional Context

Plan to fix in 3 PRs:
1. Ruff format (trivial) — #7084
2. test_refs failures (needs investigating refs.py logic regression)
3. test_model_catalog failure (needs investigating openrouter catalog change)

--- a/core/tests/test_colony_runtime_overseer.py
+++ b/core/tests/test_colony_runtime_overseer.py
@@ -274,9 +274,12 @@ class TestReportToParent:
             worker = colony.get_worker(worker_ids[0])
             assert worker is not None
 
-            # Wait for the worker's background task to finish
+            # Wait for the worker to finish AND for the SUBAGENT_REPORT event
+            # to propagate. On Windows the event loop scheduling differs from
+            # POSIX, so a worker can be marked inactive a few ticks before the
+            # subscriber callback runs. Waiting on both avoids that race.
             deadline = asyncio.get_event_loop().time() + 5.0
-            while worker.is_active and asyncio.get_event_loop().time() < deadline:
+            while (worker.is_active or len(reports) == 0) and asyncio.get_event_loop().time() < deadline:
                 await asyncio.sleep(0.05)
             assert not worker.is_active, "Worker did not finish within timeout"
 

--- a/core/tests/test_model_catalog.py
+++ b/core/tests/test_model_catalog.py
@@ -150,6 +150,11 @@ def test_openrouter_catalog_tracks_current_frontier_set():
         "anthropic/claude-opus-4.6",
         "google/gemini-3.1-pro-preview-customtools",
         "deepseek/deepseek-v3.2",
+        "qwen/qwen3.6-plus",
+        "z-ai/glm-5v-turbo",
+        "x-ai/grok-4.20",
+        "xiaomi/mimo-v2-pro",
+        "stepfun/step-3.5-flash",
     ]
     assert openrouter_models[0]["max_tokens"] == 128000
     assert openrouter_models[0]["max_context_tokens"] == 922000

--- a/tools/src/gcu/browser/refs.py
+++ b/tools/src/gcu/browser/refs.py
@@ -50,6 +50,7 @@ CONTENT_ROLES: frozenset[str] = frozenset(
         "columnheader",
         "gridcell",
         "heading",
+        "img",
         "listitem",
         "main",
         "navigation",

--- a/tools/src/gcu/browser/tools/inspection.py
+++ b/tools/src/gcu/browser/tools/inspection.py
@@ -187,8 +187,7 @@ def _resize_and_annotate(
         )
     except Exception:
         logger.warning(
-            "Screenshot resize/annotate FAILED — returning original image. "
-            "css_width=%s, dpr=%s.",
+            "Screenshot resize/annotate FAILED — returning original image. css_width=%s, dpr=%s.",
             css_width,
             dpr,
             exc_info=True,

--- a/tools/src/gcu/browser/tools/interactions.py
+++ b/tools/src/gcu/browser/tools/interactions.py
@@ -546,10 +546,7 @@ def register_interaction_tools(mcp: FastMCP) -> None:
         if x > 1.5 or y > 1.5 or x < -0.1 or y < -0.1:
             result = {
                 "ok": False,
-                "error": (
-                    f"Coords ({x}, {y}) look like pixels. This tool expects "
-                    "fractions 0..1 of the viewport."
-                ),
+                "error": (f"Coords ({x}, {y}) look like pixels. This tool expects fractions 0..1 of the viewport."),
             }
             log_tool_call("browser_hover_coordinate", params, result=result)
             return result
@@ -627,10 +624,7 @@ def register_interaction_tools(mcp: FastMCP) -> None:
         if x > 1.5 or y > 1.5 or x < -0.1 or y < -0.1:
             result = {
                 "ok": False,
-                "error": (
-                    f"Coords ({x}, {y}) look like pixels. This tool expects "
-                    "fractions 0..1 of the viewport."
-                ),
+                "error": (f"Coords ({x}, {y}) look like pixels. This tool expects fractions 0..1 of the viewport."),
             }
             log_tool_call("browser_press_at", params, result=result)
             return result

--- a/tools/tests/test_refs.py
+++ b/tools/tests/test_refs.py
@@ -45,8 +45,9 @@ class TestAnnotateSnapshot:
     def test_skips_structural_roles(self):
         annotated, ref_map = annotate_snapshot(SAMPLE_SNAPSHOT)
         roles_in_map = {entry.role for entry in ref_map.values()}
-        # navigation, main, list, listitem, paragraph are structural — no refs
-        assert "navigation" not in roles_in_map
+        # main (unnamed), list, listitem (unnamed), paragraph are structural — no refs.
+        # Note: navigation is a landmark role and now gets a ref when named, so it
+        # is not asserted absent here.
         assert "main" not in roles_in_map
         assert "list" not in roles_in_map
         assert "listitem" not in roles_in_map

